### DISCUSSION TITLE
Curved edges when using MaterialGrid

### DISCUSSION THREAD

**Discussion by ramji28** [NanoComp/meep#2760] [Q&A] [answered]
Hello Meep community,

I'm attempting to generate a square grid with two materials using `MaterialGrid.` However, when visualizing the structure using `Plot2D` the material doesn't appear as a perfect square. I've set $\\beta=\infty$. Can anyone help me identify what might be the issue?
Thank you for the help

![matgrid_graded_res70_11_11](https://github.com/NanoComp/meep/assets/87534556/d88ef48c-d7c3-408b-a966-76020e644a2b)
```ruby
import meep as mp
import numpy as np
import matplotlib
matplotlib.use('agg')
import matplotlib.pyplot as plt

resolution = 70

cell_size = mp.Vector3(2,2,0)
weights = np.ones((11,11))
weights[0,0]=0
Nx = len(weights)
Ny = len(weights)
mat_interp = mp.MaterialGrid(mp.Vector3(Nx,Ny,0),
                             mp.Medium(index=3.5),
                             mp.Medium(index=1.5),
                             weights=weights,
                             beta=mp.inf,)

geometry = [mp.Block(material=mat_interp,
                     center=mp.Vector3(),
                     size=mp.Vector3(2,2,0))]

sim = mp.Simulation(cell_size=cell_size,
                    resolution=resolution,
                    geometry=geometry)
fig, ax = plt.subplots()
sim.plot2D(ax=ax)
fig.suptitle('resolution = {}, (Nx,Ny) = ({},{})'.format(resolution,Nx,Ny))
fig.savefig('matgrid_graded_res{}_{}_{}.png'.format(resolution,Nx,Ny),
            bbox_inches='tight',
            dpi=300)
```

**Comment by peterropac** [comment]
This might happen because of subpixel smoothing. You can try `eps_averaging=False` in the `Simulation` class. Have you tried increasing the resolution of the `MaterialGrid`?

**Comment by smartalecH** [comment, marked as answer]
Why is your MaterialGrid so course? Typically you should match or exceed the resolution as your simulation. 

When you specify a beta, meep runs a linear interpolation routine to map the MaterialGrid to the yee lattice. This acts like a spatial low pass filter.

Try a much finer MaterialGrid.

**Reply by ramji28** [reply]
Thank you so much. I used a coarse grid to generate a random grid for different material shapes to train a neural network. However, as you mentioned, I got the square using a finer grid. 
![matgrid_finer_res70_71_71](https://github.com/NanoComp/meep/assets/87534556/c72071e2-4160-4c77-bf0e-e854f17e9934)
